plugin/evm/orderbook/tx_processor.go

```diff
@@ -169,47 +170,62 @@ func (lotp *limitOrderTxProcessor) executeLocalTx(contract common.Address, contr
 		log.Error("lop.txPool.AddOrderBookTx failed", "err", err, "tx", signedTx.Hash().String(), "nonce", nonce)
 		return txHash, err
 	}
-	// log.Info("executeLocalTx - AddOrderBookTx success", "tx", signedTx.Hash().String(), "nonce", nonce)
 
 	return txHash, nil
 }
 
-func (lotp *limitOrderTxProcessor) getBaseFeeEstimate() *big.Int {
-	baseFeeEstimate, err := lotp.backend.EstimateBaseFee(context.TODO())
+func (lotp *limitOrderTxProcessor) getTransactionFee() *big.Int {
+	latest := lotp.backend.CurrentHeader()
+	latestBlockNumber := latest.Number.Uint64()
+
+	// if the fee is already calculated for this block, then return it
+	if lotp.validatorTxFeeConfig.blockNumber == latestBlockNumber {
+		return lotp.validatorTxFeeConfig.baseFeeEstimate
+	}
+
+	baseFeeEstimate, err := lotp.backend.SuggestPrice(context.Background())
 	if err != nil {
-		baseFeeEstimate = big.NewInt(0).Abs(lotp.backend.CurrentBlock().BaseFee())
-		log.Error("Error in calculating updated bassFee, using last header's baseFee", "baseFeeEstimate", baseFeeEstimate)
+		log.Error("getBaseFeeEstimate - SuggestPrice failed", "err", err)
+		return big.NewInt(65_000000000) // hardcoded to 70 gwei
 	}
-	return baseFeeEstimate
-}
+	// add 10%
+	baseFeeEstimate.Add(baseFeeEstimate, big.NewInt(0).Div(baseFeeEstimate, big.NewInt(10)))
 
-func (lotp *limitOrderTxProcessor) updateValidatorTxFeeConfig() {
-	currentBlockNumber := lotp.backend.CurrentBlock().NumberU64()
-	if lotp.validatorTxFeeConfig.blockNumber < currentBlockNumber {
-		baseFeeEstimate := lotp.getBaseFeeEstimate()
-		// log.Info("inside lotp updating txFeeConfig", "blockNumber", currentBlockNumber, "baseFeeEstimate", baseFeeEstimate)
-		lotp.validatorTxFeeConfig.baseFeeEstimate = baseFeeEstimate
-		lotp.validatorTxFeeConfig.blockNumber = currentBlockNumber
+	feeConfig, _, err := lotp.backend.GetFeeConfigAt(latest)
+	if err != nil {
+		log.Error("getBaseFeeEstimate - GetFeeConfigAt failed", "err", err)
+		// if feeConfig can't be obtained, then add another 10% to the baseFeeEstimate
+		baseFeeEstimate.Add(baseFeeEstimate, big.NewInt(0).Div(baseFeeEstimate, big.NewInt(10)))
+		return baseFeeEstimate
 	}
+
+	blockGasCost := dummy.CalcBlockGasCost(
```

Comment:
how about we pessimistically assume that block is being produced in less than a second (and most of the times that will be case), and set it to `latest.BlockGasCost + feeConfig.BlockGasCostStep`.

Reply:
Yep, this can be done, and it should cover the block gas cost